### PR TITLE
ceph-volume: fix lvm batch auto with full SSDs

### PR DESCRIPTION
**Pull request by dsavineau**
The ceph-volume lvm batch --auto introduced by [1] breaks the backward
compatibility when using non rotational devices only (SSD and/or NVMe).
Those devices are reaffected as bluestore db or filestore journal
devices while we want them as data devices.

Fixes: https://tracker.ceph.com/issues/48106

[1] https://github.com/ceph/ceph/pull/34740

Signed-off-by: Dimitri Savineau <dsavinea@redhat.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by guits — APPROVED** [review]

File: src/ceph-volume/ceph_volume/devices/lvm/batch.py
Comment (jan--f) [inline src/ceph-volume/ceph_volume/devices/lvm/batch.py:373]: Sorry to be nit-picky about this, but I think in a few month time nobody will remember the reason for this comparison.

I think something more explicit would be good here.

```suggestion
        if ssd and not rotational:
            # no need for additional sorting, we'll only deploy standalone on ssds
            return
```

**Review by jan--f — CHANGES_REQUESTED** [review]
Would you mind adding a unit test for this too please? This should certainly be caught in our CI not rooks.

**Comment by dsavineau** [comment]
jenkins test ceph-volume tox

**Comment by jan--f** [comment]
jenkins test ceph-volume all

**Review by jan--f — APPROVED** [review]
Thanks guys, looks good!

**Comment by guits** [comment]
jenkins test ceph-volume lvm centos8-bluestore-dmcrypt

**Review by rishabh-d-dave — APPROVED** [review]

**Comment by dsavineau** [comment]
@jan--f @guits I was wondering if we should add more tests for `_sort_rotational_disks` because we only added one test for the full SSD scenario but I guess we should also have a test for
  - full HDD scenario
  - bluestore mixed HDD/SSD scenario (for db_devices)
  - filestore mixed HDD/SSD scenario (for journal_devices)

What do you think ?

**Comment by guits** [comment]
@dsavineau in a perfect world, we should even have 100% coverage, so yes we should add these tests.
Maybe this could be done in a dedicated PR if we want to get this fix merged quickly (although we would have to wait for a new release to include this fix anyway...)

your call... guys 🙂

**Comment by jan--f** [comment]
@dsavineau  More tests are always more than welcome! I'll merge this here tomorrow, unless you'll let me know more tests are going to be added here, so up to you.

In any case I have added https://tracker.ceph.com/issues/48150 to track the need for more tests.

**Comment by dsavineau** [comment]
jenkins test ceph-volume tox

**Comment by dsavineau** [comment]
jenkins test ceph-volume all

**Comment by jan--f** [comment]
@andrewschoen Looks like #37741 broke a tox test. Mind having a look at it?

**Comment by dsavineau** [comment]
jenkins test ceph-volume tox

**Comment by dsavineau** [comment]
jenkins test ceph-volume all

**Comment by jan--f** [comment]
jenkins test make check

**Comment by jan--f** [comment]
> @andrewschoen Looks like #37741 broke a tox test. Mind having a look at it?

Nevermind @andrewschoen  :) Thanks everyone!

**Pull request merged**
